capsules/src/net/util.rs

```diff
@@ -0,0 +1,101 @@
+/// Utility functions used in the 6LoWPAN implementation
+
+/// Verifies that all bits beyond prefix_len in the slice are zero
+pub fn verify_prefix_len(prefix: &[u8], prefix_len: u8) -> bool {
```

Comment:
Yep, `prefix_len` is in bits. The goal is to check if the provided slice is zero-padded after the prefix length (in bits) and that the slice is of the right length, so it is not quite just a single loop. I do agree that the iterator way of doing this is cleaner though, but the masking check for the remaining bits must still be done. This probably needs a more detailed comment.

```rust
pub fn verify_prefix_len(buf: &[u8], prefix_len: u8) -> bool {
    let full_bytes = (prefix_len / 8) as usize;
    let remaining_bits = prefix_len % 8;
    let bytes = full_bytes + if remaining_bits != 0 { 1 } else { 0 };
    
    if buf.len() < bytes {
        return false;
    }
    
    if remaining_bits != 0 {
        let mask = 0xff >> remaining_bits;
        if buf[full_bytes] & mask != 0 {
            return false;
        }
    }
    
    buf.iter().skip(bytes).all(|&b| b == 0)
}
```